MQTTnet.Core/Client/MqttClientQueued.cs

```diff
@@ -0,0 +1,530 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using MQTTnet.Core.Adapter;
+using MQTTnet.Core.Diagnostics;
+using MQTTnet.Core.Exceptions;
+using MQTTnet.Core.Packets;
+using MQTTnet.Core.Protocol;
+using MQTTnet.Core.Internal;
+
+namespace MQTTnet.Core.Client
+{
+    public class MqttClientQueued : IMqttClient
```

Comment:
Please do NOT just copy the whole existing MqttClient. Instead use it as a field inside your new client. You should also implement the IMqttClient interface and provide proxy-methods to the internal MqttClient. But still having a different implementation for publish, the receive event etc.

Reply:
That's what I would have done, but most methods are private, for example how to use AuthenticateAsync without copy it ? we nee an abstract base class, do we?

Reply:
What is the reason for using _AuthenticateAsync_? Sorry but I don't get it.

Reply:
It was an example. To be more precise, I need to use the SendeAndReceivePackets but its Private,